### ISSUE TITLE
Authentication stuck on loading: "kind Authentication is the same as Authentication"

### ISSUE THREAD

**Issue by caasion** [shiftkey/desktop#714]
### The problem

Whenever I try to sign in to GitHub through the desktop app, it gets stuck on in a constant spinning state. I am able to log in through my web browser (FireFox) but it doesn't do anything after redirecting. I have tried using the app image and deb, but both do the same thing.

### Release version

3.0.0

### Operating system

Ubuntu 20.04

### Steps to reproduce the behavior

1. Download the app from .deb file or appimage
2. sign in
3. enter credentials in browser
4. redirect auth back to the desktop app
5. nothing happens

### Log files

```
2022-05-09T01:30:16.402Z - info: [ui] [AppStore] loading 0 repositories from store
2022-05-09T01:30:16.484Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:30:16.486Z - info: [ui] launching: 3.0.0-linux1 (Linux 5.15.0-27-generic)
2022-05-09T01:30:16.486Z - info: [ui] execPath: '/tmp/.mount_GitHubNkuSD1/github-desktop'
2022-05-09T01:30:16.517Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:30:16.546Z - info: [ui] Opt in reported.
2022-05-09T01:30:16.631Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:30:21.014Z - info: [ui] [Welcome] advancing to step: SignInToDotComWithBrowser
2022-05-09T01:30:21.015Z - info: [ui] [SignInStore] initializing OAuth flow
2022-05-09T01:30:21.021Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=3a723b10ac5575cc5bb9&scope=repo%20user%20workflow&state=253a6712-3ff0-432c-85cd-143eca241709
2022-05-09T01:30:21.022Z - info: [ui] [Welcome] current sign in state not found. ignoring...
2022-05-09T01:30:21.414Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:31:13.941Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:31:17.083Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:31:57.923Z - info: [ui] [AppStore] loading 0 repositories from store
2022-05-09T01:31:57.996Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:31:58.001Z - info: [ui] launching: 3.0.0-linux1 (Linux 5.15.0-27-generic)
2022-05-09T01:31:58.001Z - info: [ui] execPath: '/tmp/.mount_GitHubak0jLn/github-desktop'
2022-05-09T01:31:58.084Z - info: [ui] Opt in reported.
2022-05-09T01:31:58.174Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:31:59.948Z - info: [ui] [Welcome] advancing to step: SignInToDotComWithBrowser
2022-05-09T01:31:59.950Z - info: [ui] [SignInStore] initializing OAuth flow
2022-05-09T01:31:59.956Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=3a723b10ac5575cc5bb9&scope=repo%20user%20workflow&state=331acd64-042c-4ad3-b653-387b4838e666
2022-05-09T01:31:59.958Z - info: [ui] [Welcome] current sign in state not found. ignoring...
2022-05-09T01:32:00.289Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:32:23.772Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:32:31.231Z - info: [ui] [Welcome] advancing to step: ConfigureGit
2022-05-09T01:33:01.472Z - info: [ui] [SignInStore] initializing OAuth flow
2022-05-09T01:33:01.477Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=3a723b10ac5575cc5bb9&scope=repo%20user%20workflow&state=2094fa28-9ab2-4a32-a099-dc01686b129b
2022-05-09T01:33:40.116Z - info: [ui] [SignInStore] initializing OAuth flow
2022-05-09T01:33:40.128Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=3a723b10ac5575cc5bb9&scope=repo%20user%20workflow&state=c86b9427-b247-40c8-bdf1-1783a005bd91
2022-05-09T01:36:36.055Z - info: [ui] [AppStore] loading 0 repositories from store
2022-05-09T01:36:36.133Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:36:36.136Z - info: [ui] launching: 3.0.0-linux1 (Linux 5.15.0-27-generic)
2022-05-09T01:36:36.137Z - info: [ui] execPath: '/tmp/.mount_GitHublTYZnD/github-desktop'
2022-05-09T01:36:36.170Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:36:36.265Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:36:36.538Z - info: [ui] Opt in reported.
2022-05-09T01:37:00.991Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:37:41.775Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:37:41.902Z - info: [ui] [Welcome] advancing to step: SignInToDotComWithBrowser
2022-05-09T01:37:41.903Z - info: [ui] [SignInStore] initializing OAuth flow
2022-05-09T01:37:41.907Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=3a723b10ac5575cc5bb9&scope=repo%20user%20workflow&state=26f0273f-608b-4c66-a07e-abf4c6d33031
2022-05-09T01:37:41.908Z - info: [ui] [Welcome] current sign in state not found. ignoring...
2022-05-09T01:37:42.225Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:39:51.678Z - info: [main] Received possible protocol arguments: 3
2022-05-09T01:39:53.018Z - info: [ui] [AppStore] loading 0 repositories from store
2022-05-09T01:39:53.063Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:39:53.065Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:39:53.106Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:39:53.117Z - info: [ui] launching: 3.0.0-linux1 (Linux 5.15.0-27-generic)
2022-05-09T01:39:53.118Z - info: [ui] execPath: '/tmp/.mount_GitHubSkB1fE/github-desktop'
2022-05-09T01:39:53.143Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:39:53.187Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:39:53.257Z - info: [ui] Opt in reported.
2022-05-09T01:39:53.523Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:39:54.764Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:39:54.898Z - info: [ui] [Welcome] advancing to step: SignInToDotComWithBrowser
2022-05-09T01:39:54.900Z - info: [ui] [SignInStore] initializing OAuth flow
2022-05-09T01:39:54.904Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=3a723b10ac5575cc5bb9&scope=repo%20user%20workflow&state=4361db28-f06e-4617-b596-53cb2bbad2ff
2022-05-09T01:39:54.908Z - info: [ui] [Welcome] current sign in state not found. ignoring...
2022-05-09T01:39:55.209Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:40:40.284Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:40:48.326Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:41:45.122Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:41:45.906Z - info: [ui] [Welcome] advancing to step: Start
2022-05-09T01:41:47.807Z - info: [main] opening in browser: https://github.com/join?source=github-desktop
2022-05-09T01:41:48.100Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:41:54.751Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:41:54.872Z - info: [ui] [Welcome] advancing to step: SignInToDotComWithBrowser
2022-05-09T01:41:54.873Z - info: [ui] [SignInStore] initializing OAuth flow
2022-05-09T01:41:54.877Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=3a723b10ac5575cc5bb9&scope=repo%20user%20workflow&state=8c6f7d57-9776-48d5-8f0c-54f8e6abc549
2022-05-09T01:41:54.878Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:41:55.219Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:42:48.985Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:42:50.713Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:43:38.395Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
2022-05-09T01:43:39.314Z - info: [ui] [Welcome] advancing to step: Start
2022-05-09T01:46:59.992Z - info: [main] Received possible protocol arguments: 1
2022-05-09T01:51:19.600Z - info: [ui] [AppStore] loading 0 repositories from store
2022-05-09T01:51:19.673Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:51:19.674Z - info: [ui] launching: 3.0.0-linux1 (Linux 5.15.0-27-generic)
2022-05-09T01:51:19.674Z - info: [ui] execPath: '/usr/lib/github-desktop/github-desktop'
2022-05-09T01:51:19.721Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:51:19.902Z - info: [ui] Opt in reported.
2022-05-09T01:51:20.642Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:51:23.643Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:51:23.676Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:51:39.264Z - info: [ui] [Welcome] no sign in step found. ignoring...
2022-05-09T01:51:40.758Z - info: [ui] [Welcome] advancing to step: SignInToDotComWithBrowser
2022-05-09T01:51:40.760Z - info: [ui] [SignInStore] initializing OAuth flow
2022-05-09T01:51:40.765Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=3a723b10ac5575cc5bb9&scope=repo%20user%20workflow&state=dfc22bd7-4eb0-4931-8ff2-b5eb82b1d996
2022-05-09T01:51:40.767Z - info: [ui] [Welcome] current sign in state not found. ignoring...
2022-05-09T01:51:41.077Z - info: [ui] [Welcome] kind Authentication is the same as Authentication. ignoring...
```

**Comment by TaromaruYuki** [comment]
What worked for me is following issue: #632 

TLDR: Install 2.9.6, sign in, then upgrade to latest.

**Comment by shiftkey** [comment]
> 3. enter credentials in browser
> 4. redirect auth back to the desktop app

The bit that's missing here is Firefox using the protocol handlers we register from the `deb` to pass the callback URL to the app. There's a couple of messages that indicate something is happening but we don't print the raw details because this is sensitive information:

```
2022-05-09T01:39:51.678Z - info: [main] Received possible protocol arguments: 3
...
2022-05-09T01:46:59.992Z - info: [main] Received possible protocol arguments: 1
```

These are the checks that we do to find the important text:

https://github.com/shiftkey/desktop/blob/64fab85c48ae3b69c97569913803a5400893c079/app/src/main-process/main.ts#L265-L276

Because we'd get a lot of extra noise from the parameters that was not important for the app:

```
handlePossibleProtocolLauncherArgs: ["github-desktop","--enable-crashpad","--allow-file-access-from-files","x-github-desktop-dev-auth://oauth?code=[snip]&state=[snip]"]
```

I suspect this has changed underneath me again, but I'm not sure when I'll be able to dig into this to see if I can reproduce it on my end...

**Comment by blounsbury** [comment]
Same issue here. Using Ubuntu 22.04 with firefox and chromium, tried v2.6.6 and v2.9.6 which don't work at all. The newest release loads and authenticates to github and redirects to github desktop and nothing happens. Can't get into github desktop. Hoping this can get addressed, soon?

**Comment by shiftkey** [comment]
I feel like this has been stuck for a while and was likely fixed elsewhere, so either testing again with a newer release or providing more information here is the next step forward...

**Comment by blounsbury** [comment]
Sorry, not trying to reopen this or anything. Just confirming that I tried again just now, and everything works perfectly! Thank you!
